### ISSUE TITLE
`__getattr__` is unsafe on subclasses

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#880]
Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&reportUninitializedInstanceVariable=false&code=MYGwhgzhAEBiD28BcBYAUNT0AmBTAZtAPpEDmuALmBRQE4kAUEuI%2BANNAHZgC2uS0CHQCU0ALQA%2BaAEtOFVBixLalAK61O0AIzp0oSDABCYWgwTxhCpZGz4BQ2tAC80AESvdaPIXycGANzAQVX44REt0ZVx-XCCiCgBPAAdcAKCQgDobfFEAYhk5DgAjVQpBOmhqaFpVOWk%2BT18GY1NhYSA)

```python
class Foo:
    def __getattr__(self, name: str) -> int:
        return 1

class Bar(Foo):
    asdf: str = ""

def fn(value: Foo):
    reveal_type(value.asdf) # int, but str at runtime

fn(Bar())
```

when subtyping a class with a `__getattr__`, it should report an error on any attributes on the subclass that are not compatible with the return type of `__getattr__` (unless it's an override of an attribute that exists on the superclass)
